src/main/java/upb/dice/twitter/IDHandler.java

```diff
@@ -0,0 +1,131 @@
+package upb.dice.twitter;
+
+import twitter4j.GeoLocation;
+import twitter4j.Query;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Stores and retrieves the maxID, oldestTweetID and the sinceID with the respective keyword/location for a search result which is used for the query generator
+ * during the next search
+ */
+public class IDHandler {
+    private static String directoryName = "Tweets_Search_Details";
+    private static String keywordMaxIDFilepath = directoryName + File.separator + "Keyword_MaxID1.txt";
+    private static String locationMaxIDFilepath = directoryName + File.separator + "Location_MaxID1.txt";
+
+    /**
+     * This method stores the maxID depending on the query
+     *
+     * @param maxID the lower bound for the further queries
+     * @param query determining the location based or keyword based
+     */
+    public void storeMaxID(double maxID, Query query) throws IOException {
+        //keywordBased MaxID
+        if (query.getQuery() != null) {
+            try (BufferedWriter out = new BufferedWriter(new FileWriter(new File(keywordMaxIDFilepath), true))) {
+                out.append(query.getQuery()).append(",").append(String.format("%.0f", maxID));
+            }
+        } else {
+            try (BufferedWriter out = new BufferedWriter(new FileWriter(new File(locationMaxIDFilepath), true))) {
+                out.append(String.valueOf(query.getGeocode())).append(",").append(String.format("%.0f", maxID));
+            }
+        }
+    }
```

Comment:
Having three different methods that are writing the three different IDs is resource demanding and error prone. Why are you not simply using a simple class that stores the current "state" of a query (i.e., the three ids)? In that case, you simply have one single method for storing a state and a second for loading a state for a given keyword.